src/toolkits/style_transfer/style_transfer_data_iterator.hpp

```diff
@@ -0,0 +1,96 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef __TOOLKITS_STYLE_TRANSFER_H_
+#define __TOOLKITS_STYLE_TRANSFER_H_
+
+#include <functional>
+#include <map>
+#include <memory>
+
+#include <core/data/sframe/gl_sframe.hpp>
+#include <core/logging/table_printer/table_printer.hpp>
+#include <ml/neural_net/compute_context.hpp>
+#include <ml/neural_net/model_backend.hpp>
+#include <ml/neural_net/model_spec.hpp>
+#include <model_server/lib/extensions/ml_model.hpp>
+#include <toolkits/coreml_export/mlmodel_wrapper.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+struct st_image {
+  image_type style;
+  image_type content;
+  size_t index;
```

Comment:
I'm going to obsess over names, because I think names are important.

For the two images, could we go with a more complete name, like `style_image` and `content_image`? And I presume `index` is `style_index`?

Maybe `st_image` could be `st_example`?

This what we get more readable code like:
```
st_example example = ...;
do_stuff(example.style_image, example.content_image);
```